README.md

```diff
@@ -118,16 +108,48 @@ Fired when the scroller has finished rendering a new range of items, e.g. becaus
 
 Also see [the example below](#performing-actions-as-the-scroller-scrolls-using-the-rangechange-event).
 
+### The `ItemSource` class
+
+The `ItemSource` class represents a way of translating indices into JavaScript values. You can create them like so:
+
+```js
+const source = new ItemSource({

```

Comment:
Is there a need to have a class? Implementation-wise, I feel this is not really needed. Also it looks more ergonomic to either accept an object or an array, e.g.
```js
// either object...
scroller.itemSource = {
  getLength() {},
  item(index) {},
  key(index) {}
};
// ...or array
scroller.itemSource = new Array(100);
```

Reply:
Any reason getLength is not a properly like Array.length?

Reply:
I think we need a class so that `list.itemSource` can return something useful and encapsulated and the same type each time.

Reply:
Also it allows us to have a simple factory, `ItemSource.fromArray(x, keyFunc)`, as well as potentially future other factories.